### PR TITLE
feat: Add research keywords

### PR DESCRIPTION
**Pull request by tim-vd-aardweg**
refs: #634

### CODE REVIEW

**Comment by tim-vd-aardweg** [comment]
I wanted to add this test:

```python
    def test_load_model_with_research_keywords_as_fmmodel_raises_error(self):
        input_mdu = (
                test_input_dir / "research" / "mdu_with_research_keywords_from_dia_file_2024.03_release.mdu"
        )

        with pytest.raises(ValueError) as e:
            _ = FMModel(filepath=input_mdu)

        expected_error = "Unknown keywords are detected in section"
        assert expected_error in str(e.value)
```

But that test can only pass once #632 is merged. So for now I decided to keep this test out.

File: hydrolib/core/dflowfm/research/models.py [hunk 1]
```diff
@@ -0,0 +1,903 @@
...
+    numerics: ResearchNumerics = Field(default_factory=ResearchNumerics)
+    physics: ResearchPhysics = Field(default_factory=ResearchPhysics)
+    sediment: ResearchSediment = Field(default_factory=ResearchSediment)
+    wind: ResearchWind = Field(default_factory=ResearchWind)
+    waves: ResearchWaves = Field(default_factory=ResearchWaves)
+    time: ResearchTime = Field(default_factory=ResearchTime)
+    restart: ResearchRestart = Field(default_factory=ResearchRestart)
+    trachytopes: ResearchTrachytopes = Field(default_factory=ResearchTrachytopes)
+    output: ResearchOutput = Field(default_factory=ResearchOutput)
+    processes: Optional[ResearchProcesses] = Field(None)
```
Comment (MRVermeulenDeltares) [inline hydrolib/core/dflowfm/research/models.py:903]: is this Field(None) correct?

**Review by MRVermeulenDeltares — APPROVED** [review]

File: hydrolib/core/dflowfm/research/models.py (hunk 1 above)
Comment (tim-vd-aardweg) [reply hydrolib/core/dflowfm/research/models.py:903]: Yes. The [processes] section is Optional.

**Review by MRVermeulenDeltares — CHANGES_REQUESTED** [review]
As discussed the research_ also need to be added for comments.

**Review by MRVermeulenDeltares — APPROVED** [review]

**Pull request merged**